File: solutions/go/cars-assemble/1/cars_assemble.go
package cars

// The size of batches
const batchSize = 10
// The price to produce batchSize cars
const batchPrice = 95000
// The price to produce one car
const unitPrice = 10000

func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * (successRate / 100.0)
}


func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(CalculateWorkingCarsPerHour(productionRate, successRate)) / 60
}

func CalculateCost(carsCount int) uint {
	quotient, remainder := carsCount/10, carsCount%10
    return uint(quotient*batchPrice + remainder*unitPrice)

}

File: solutions/lua/cars-assemble/1/cars-assemble.lua
local cars = {}

-- returns the amount of working cars produced by the assembly line every hour
function cars.calculate_working_cars_per_hour(production_rate, success_rate)
  return (production_rate*success_rate) / 100
end

-- returns the amount of working cars produced by the assembly line every minute
function cars.calculate_working_cars_per_minute(production_rate, success_rate)
  return cars.calculate_working_cars_per_hour(production_rate, success_rate) // 60
end

cars.cost_per_car_per_qty = {}
cars.cost_per_car_per_qty[10] = 95000
cars.cost_per_car_per_qty[1] = 10000
-- returns the cost of producing the given number of cars
function cars.calculate_cost(cars_count)
  return cars.cost_per_car_per_qty[1] * (cars_count % 10) + cars.cost_per_car_per_qty[10] * (cars_count // 10)
end

return cars
